src/data/index.ts

```diff
@@ -7,11 +7,32 @@ import { Memento } from 'vscode';
 import { PersistPartial } from 'redux-persist/es/persistReducer';
 import { persistedStateCodecNew } from '../persistedState/codecs';
 
-const buildStore = (workspaceState: Memento) => {
+const PERSISTANCE_PREFIX = 'persist';
+const PERSISTANCE_KEY = 'compressedRoot';
+
+const deserializeState = (serializedState: string) => {
+	const parsedState: Record<string, unknown> = {};
+
+	try {
+		const rawState = JSON.parse(serializedState);
+
+		Object.keys(rawState).forEach((key) => {
+			parsedState[key] = JSON.parse(rawState[key]);
```

Comment:
hmm, can you explain why that's needed?

Reply:
this is needed because `redux-persist` stores each property as a string, so you need to parse the state object, and then parse each field

Reply:
link: https://github.com/rt2zz/redux-persist/blob/d8b01a085e3679db43503a3858e8d4759d6f22fa/src/getStoredState.ts#L28

Reply:
@DmytroHryshyn does it mean we need to do this recursively, or just one level?

Reply:
only one level